includes/wc-product-functions.php

```diff
@@ -1454,6 +1465,38 @@ function wc_update_product_lookup_tables_column( $column ) {
 				"
 			);
 			break;
+		case 'onsale':
+			$column   = esc_sql( $column );
+			$decimals = absint( wc_get_price_decimals() );
+
+			$wpdb->query(
```

Comment:
Same as the other suggestion, do we need to do this query prior to setting the value with another update statement?

Reply:
The first resets all values to 0. The 2nd only sets values to 1. If this process is re-ran, without the first you'd only set rows to true, not false.